src/prefect/agent/kubernetes/agent.py

```diff
@@ -191,6 +193,58 @@ def manage_jobs(self) -> None:
                                         delete_job = True
                                         break
 
+                    # Report failed pods
+                    if job.status.failed:
+                        pods = self.core_client.list_namespaced_pod(
+                            namespace=self.namespace,
+                            label_selector="prefect.io/identifier={}".format(
+                                job.metadata.labels.get("prefect.io/identifier")
+                            ),
+                        )
+
+                        failed_pods = []
+                        for pod in pods.items:
+                            if pod.status.phase != "Failed":
+                                continue
+
+                            failed_pods.append(pod.metadata.name)
+
+                            for status in pod.status.container_statuses:
+                                if status.state.terminated:
```

Comment:
Not all terminated containers are indications of failure - if a user has a sidecar container for a pod that exited successfully that doesn't necessarily need to be logged.

Perhaps instead of writing one log record for each container in a pod, we write one log record that includes info on all containers in the pod?

We might want to also include only non-null fields (everything but `exit_code` below is optional and may be null.

So something like:

```
Pod <pod name> failed.
  Container <container name>:
    all non-null fields as you have them
  Container <container name>:
    ...
```

I think including successful containers seems fine (more information for the user), but like not having an ERROR log that only mentions a successful container (grouping the log message together fixes that, since one of those containers will be a failure).